chimney/src/main/scala/io/scalaland/chimney/internal/macros/TransformerMacros.scala

```diff
@@ -619,10 +619,21 @@ trait TransformerMacros extends TransformerConfigSupport with MappingMacros with
     } else {
       val overridesMapping = resolveOverrides(srcPrefixTree, From, targets, config)
       val notOverridenTargets = targets.diff(overridesMapping.keys.toSeq)
-      val accessorsMapping = resolveAccessorsMapping(From, notOverridenTargets, config)
 
-      resolveTransformerBodyTreeFromAccessorsMapping(srcPrefixTree, accessorsMapping, From, To, config)
-        .map(_ ++ overridesMapping)
+      def mkTransformer(From: Type, srcPrefixTree: Tree) = {
+        val accessorsMapping = resolveAccessorsMapping(From, notOverridenTargets, config)
+
+        resolveTransformerBodyTreeFromAccessorsMapping(srcPrefixTree, accessorsMapping, From, To, config)
+          .map(_ ++ overridesMapping)
+      }
+
+      /** Proto oneOf to sealed trait enum support
+       *  @see https://github.com/wix-private/server-infra/issues/14909
+       */
+      if (From.baseClasses.exists(_.fullName.contains("scalapb.GeneratedOneof")))
```

Comment:
Cool, this works when converting from `oneOf` to domain types. Looks like converting vice versa would be as easy, right? 😸

Reply:
yeah, should be straightforward
I'm wondering if Piotr was ok with merging this 💩 upstream 🤣